### PR TITLE
Fixes a playing card runtime

### LINKED ISSUE #19447: Deck of cards runtime
[2022-10-19T20:43:02] Runtime in cards.dm,419: Cannot read null.l_hand
   proc name: IsAvailable (/datum/action/item_action/remove_card/IsAvailable)
   usr: Ali Zadovsky (generaldonothing) (/mob/living/carbon/human)
   usr.loc: The floor (111,141,2) (/turf/simulated/floor/plasteel)
   src: Remove a card - Remove a singl... (/datum/action/item_action/remove_card)
   call stack:
   Remove a card - Remove a singl... (/datum/action/item_action/remove_card): IsAvailable()
   Remove a card - Remove a singl... (/datum/action/item_action/remove_card): UpdateButtonIcon()
   the a playing card (/obj/item/cardhand): update appearance(11)
   the suspicious looking deck of... (/obj/item/deck/cards/syndicate): deal at(Ali Zadovsky (/mob/living/carbon/human), Ali Zadovsky (/mob/living/carbon/human), 10)
   the suspicious looking deck of... (/obj/item/deck/cards/syndicate): Deal Multiple Cards()
   Deal multiple card - Deal mult... (/datum/action/item_action/deal_card_multi): Trigger()
   Deal multiple card - Deal mult... (/obj/screen/movable/action_button): Click(null, "mapwindow.map", "icon-x=12;icon-y=25;left=1;but...")

**Steps to reproduce the problem**:
1. Have your deck of card in your active hand
2. Deal more than one card
3. runtimeeeee

**When did the problem start happening**:
likely forever

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
fixes #19447 

## Why It's Good For The Game
I don't think this code actually does anything; no owner would be set at the point of addition, so it probably doesn't need to be there

## Testing
compiled, ran, tried to break it and failed
## Changelog
NPFC

### CODE REVIEW

File: code/modules/games/cards.dm [hunk 1]
```diff
@@ -415,9 +415,6 @@
 	var/obj/item/cardhand/C = target
 	if(length(C.cards) <= 1)
 		return FALSE
-	var/mob/living/carbon/human/O = owner
-	if(O.l_hand && O.r_hand)
```
Comment (S34NW) [inline code/modules/games/cards.dm:419]: I _think_ what this is trying to do is make the action unuseable if you have something in both your hands, right?

Comment (GDNgit) [reply code/modules/games/cards.dm:419]: it doesn't

**Review by S34NW — APPROVED** [review]

File: code/modules/games/cards.dm (hunk 1 above)
Comment (Sirryan2002) [reply code/modules/games/cards.dm:419]: can you check to make sure that this doesn't break the `remove_card` action. I believe this is intended to make sure that the user doesn't just remove the card onto the ground if they have nothing in their hands.

Comment (GDNgit) [reply code/modules/games/cards.dm:419]: > can you check to make sure that this doesn't break the `remove_card` action. I believe this is intended to make sure that the user doesn't just remove the card onto the ground if they have nothing in their hands.

fixed

File: code/modules/games/cards.dm [hunk 2]
```diff
@@ -462,6 +462,9 @@
 		return
 
 	var/datum/playingcard/card = pickablecards[pickedcard]
+	var/picked_card_loc = src.loc
+	if(picked_card_loc != user) // Don't want people teleporting cards
```
Comment (Sirryan2002) [inline code/modules/games/cards.dm:466]:
```suggestion
	if(loc != user) // Don't want people teleporting cards
```

File: code/modules/games/cards.dm [hunk 3]
```diff
@@ -499,6 +502,10 @@
 		if(!discarding)
 			continue
 
+		var/picked_card_loc = src.loc
+		if(picked_card_loc != user) // Don't want people teleporting cards
```
Comment (Sirryan2002) [inline code/modules/games/cards.dm:506]:
```suggestion
		if(loc != user) // Don't want people teleporting cards

File: code/modules/games/cards.dm (hunk 1 above)
Comment (Sirryan2002) [reply code/modules/games/cards.dm:419]: can you check to make sure that if the users hands are full, that they don't just drop the card on the ground?

File: code/modules/games/cards.dm (hunk 3 above)
Comment (GDNgit) [reply code/modules/games/cards.dm]: applied

File: code/modules/games/cards.dm (hunk 2 above)
Comment (GDNgit) [reply code/modules/games/cards.dm]: applied

**Review by Sirryan2002 — APPROVED** [review]
code looks fine

**Pull request merged**